sdk/include/opentelemetry/sdk/metrics/state/sync_metric_storage.h

```diff
@@ -17,60 +23,132 @@ class SyncMetricStorage : public MetricStorage, public WritableMetricStorage
 {
 
 public:
-  SyncMetricStorage(
-      // TBD
-      // Aggregation& aggregation
-      // opentelemetry::sdk::resource::resource *resource,
-      // sdk::instrumentationlibrary::InstrumentationLibrary *instrumentation_library,
-  )
-  {}
+  SyncMetricStorage(InstrumentDescriptor instrument_descriptor,
+                    const AggregationType aggregation_type,
+                    AttributesProcessor *attributes_processor = new DefaultAttributesProcessor())
+      : aggregation_type_{aggregation_type},
+        instrument_descriptor_(instrument_descriptor),
+        attributes_processor_{attributes_processor},
+        attributes_hashmap_(std::move(std::unique_ptr<AttributesHashMap>(new AttributesHashMap)))
```

Comment:
This can be simplified to:
`attributes_hashmap_(new AttributesHashMap)`

Reply:
Yes correct, seems I overengineered it :)